### DISCUSSION TITLE
Building an object in a local coordinate system

### DISCUSSION THREAD

**Discussion by kurt-rhee** [compas-dev/compas#1082] [Q&A]
Hello, 

I was wondering if it would be possible to create an object in a local frame and then convert it once it is done to the world frame.  For example would it be possible to create a polygon:
```
    polygon = Polygon([
        [0, 0, 0],
        [1, 0, 0],
        [1, 1, 0],
        [0, 1, 0]
    ])
```

I would like this polygon to be situated at the origin in the local frame, but translated in the world frame.

**Comment by kurt-rhee** [comment]
```
    local_coordinate_system = Frame(
        point=bay_origin,
        xaxis=bay_x_axis,
        yaxis=bay_y_axis
    )
    to_lcs = Transformation.from_frame_to_frame(
        Frame.worldXY(),
        local_coordinate_system
    )

    # --- bay polygon object ---
    # create a 2d bay object centered on the torque tube
    polygon = Polygon([
            [0, 0, 0],
            [1, 0, 0],
            [1, 1, 0],
            [0, 1, 0]
        ])
    polygon.transform(to_lcs)

```

From what I've seen, I build an object in the world coordinate system and then send it as a transformation into the local coordinate system

**Reply by tomvanmele** [reply]
if i understand correctly, what you're looking for is this...
```python
from compas.geometry import Polygon
from compas.geometry import Frame
from compas.geometry import Transformation

from compas_view2.app import App

polygon = Polygon([[0, 0, 0], [1, 0, 0], [1, 1, 0], [0, 1, 0]])
frame = Frame([1, 1, 0], [1, 0, 0], [0, 0, 1])
transformation = Transformation.from_frame(frame)

polygon.transform(transformation)

viewer = App()
viewer.add(polygon)
viewer.add(frame, size=0.5, linewidth=5)
viewer.run()
```
<img width="1840" alt="Screenshot 2022-10-07 at 10 05 59" src="https://user-images.githubusercontent.com/29653489/194504578-412041a7-2afd-4f55-9643-2524a892000a.png">

**Comment by kurt-rhee** [comment]
Hello,

Thank you, I understand now.

Best,


Kurt

Message ID: ***@***.***
.com>
